From ee91c131a32cdb218273c761bfd7a6642f123353 Mon Sep 17 00:00:00 2001
From: Robby Cai <robby.cai@nxp.com>
Date: Wed, 22 Nov 2017 13:38:15 +0800
Subject: [PATCH 2884/5242] MLK-16919-9 dts: add MIPI CSI2 support on iMX8MQ

commit  81d4601b7e9e05d38001cf2702400047f10b729a from
https://source.codeaurora.org/external/imx/linux-imx.git

on EVK board, the camera on MIPI CSI1 and CSI2 port shares the same I2C bus
and the camera has the same slave address. So it's not possible to enable
two ports at same time. The first port is enabled by default. Here to create
a separate dts file to enable second port.

Signed-off-by: Robby Cai <robby.cai@nxp.com>
Reviewed-by: Sandor Yu <Sandor.yu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/freescale/Makefile             |    3 +-
 .../dts/freescale/fsl-imx8mq-evk-mipi-csi2.dts     |   38 ++++++++++++++++++++
 2 files changed, 40 insertions(+), 1 deletion(-)
 create mode 100644 arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-mipi-csi2.dts

diff --git a/arch/arm64/boot/dts/freescale/Makefile b/arch/arm64/boot/dts/freescale/Makefile
index 81fc07c..05ffa50 100644
--- a/arch/arm64/boot/dts/freescale/Makefile
+++ b/arch/arm64/boot/dts/freescale/Makefile
@@ -51,4 +51,5 @@ dtb-$(CONFIG_ARCH_FSL_IMX8MQ) += fsl-imx8mq-evk.dtb \
 				 fsl-imx8mq-evk-m4.dtb \
 				 fsl-imx8mq-evk-pcie1-m2.dtb \
 				 fsl-imx8mq-evk-lcdif-adv7535.dtb \
-				 fsl-imx8mq-evk-lcdif-rm67191.dtb
+				 fsl-imx8mq-evk-lcdif-rm67191.dtb \
+				 fsl-imx8mq-evk-mipi-csi2.dtb
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-mipi-csi2.dts b/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-mipi-csi2.dts
new file mode 100644
index 0000000..7e4a672
--- /dev/null
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-mipi-csi2.dts
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2017 NXP
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+#include "fsl-imx8mq-evk.dts"
+
+&ov5640_mipi {
+	status = "disabled";
+};
+
+&ov5640_mipi2 {
+	status = "okay";
+};
+
+&csi1_bridge {
+	status = "disabled";
+};
+
+&csi2_bridge {
+	status = "okay";
+};
+
+&mipi_csi_1 {
+	status = "disabled";
+};
+&mipi_csi_2 {
+	status = "okay";
+};
-- 
1.7.9.5

